Sources/Helpers/DirectoryObserver.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  DirectoryObserver.swift
+//  OsmAnd Maps
+//
+//  Created by Skalii on 26.07.2024.
+//  Copyright © 2024 OsmAnd. All rights reserved.
+//
+
+import Foundation
+
+@objc enum DirectoryObserverType: Int {
+    case colorPalette
+
+    var path: String {
+        switch self {
+        case .colorPalette:
+            return OsmAndApp.swiftInstance().colorsPalettePath
+        }
+    }
+
+    var notificationName: NSNotification.Name {
+        switch self {
+        case .colorPalette:
+            return NSNotification.Name("ColorPaletteDicrectoryUpdated")
+        }
+    }
+}
+
+@objcMembers
+final class DirectoryObserver: NSObject {
+
+    static let updatedKey = "updated"
+    static let deletedKey = "deleted"
+    static let createdKey = "created"
+
+    private let type: DirectoryObserverType
+
+    private var directoryFileDescriptor: CInt = -1
+    private var dispatchSource: DispatchSourceFileSystemObject?
+    private let queue = DispatchQueue(label: "DirectoryObserverQueue", attributes: .concurrent)
```

Comment:
Isn't global queue enough?

Reply:
I propose that we use our own queue for this purpose. It will function as a separate service. This will prevent the situation where the global queue might be occupied when we expect to receive a notification about a change in the directory